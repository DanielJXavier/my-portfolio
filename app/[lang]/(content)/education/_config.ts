export const education = [
  {
    schoolId: "estacio",
    schoolName: "Est√°cio",
    years: "2020 ~ 2021",
  },
  {
    schoolId: "fumec",
    schoolName: "Universidade FUMEC",
    years: "2010 ~ 2014",
  },
] as const;
