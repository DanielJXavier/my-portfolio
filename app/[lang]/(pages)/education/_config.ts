export const education = [
  {
    schoolId: "estacio",
    schoolName: "Est√°cio",
    startYear: "2020",
    endYear: "2021",
  },
  {
    schoolId: "fumec",
    schoolName: "Universidade FUMEC",
    startYear: "2010",
    endYear: "2014",
  },
] as const;
