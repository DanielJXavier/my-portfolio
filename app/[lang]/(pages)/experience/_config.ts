export interface ExperienceInterface {
  key: string;
  companyId: string;
  companyName: string;
  startDate: string;
  endDate: string | null;
  mainStack: string[];
  resume: boolean;
  resumeAccomplishments: number[];
  hasBlackLogo: boolean;
}

export const experience: ExperienceInterface[] = [
  {
    key: "danielxavier-1",
    companyId: "danielxavier",
    companyName: "<DanielXavier />",
    startDate: "05-2024",
    endDate: null,
    mainStack: [
      "react",
      "next",
      "tailwind",
      "node",
      "ts",
      "jest",
      "cypress",
      "html",
      "css",
      "js",
      "contentful",
      "docker",
      "vercel",
    ],
    resume: true,
    resumeAccomplishments: [1, 2, 3],
    hasBlackLogo: true,
  },
  {
    key: "pdsacademy-1",
    companyId: "pdsacademy",
    companyName: "PDSAcademy",
    startDate: "06-2025",
    endDate: "06-2025",
    mainStack: [
      "react",
      "next",
      "tailwind",
      "html",
      "css",
      "js",
      "functional-programming",
    ],
    resume: true,
    resumeAccomplishments: [1, 2, 3],
    hasBlackLogo: true,
  },
  {
    key: "olx-4",
    companyId: "olx",
    companyName: "OLX",
    startDate: "01-2024",
    endDate: "04-2024",
    mainStack: [
      "react",
      "node",
      "kong",
      "s3",
      "firebase",
      "android",
      "kotlin",
      "ios",
      "swift-ui",
      "rn",
      "aws",
    ],
    resume: true,
    resumeAccomplishments: [1, 2, 4],
    hasBlackLogo: false,
  },
  {
    key: "olx-3",
    companyId: "olx",
    companyName: "OLX",
    startDate: "07-2023",
    endDate: "12-2023",
    mainStack: [
      "python",
      "es",
      "react",
      "storybook",
      "s3",
      "androind",
      "kotlin",
      "ios",
      "swift-ui",
      "rn",
      "postgre",
      "aws",
    ],
    resume: true,
    resumeAccomplishments: [1, 2, 4, 5],
    hasBlackLogo: false,
  },
  {
    key: "olx-2",
    companyId: "olx",
    companyName: "OLX",
    startDate: "09-2022",
    endDate: "06-2023",
    mainStack: [
      "react",
      "storybook",
      "s3",
      "android",
      "kotlin",
      "ios",
      "swift-ui",
      "aws",
    ],
    resume: true,
    resumeAccomplishments: [1, 3],
    hasBlackLogo: false,
  },
  {
    key: "olx-1",
    companyId: "olx",
    companyName: "OLX",
    startDate: "08-2021",
    endDate: "08-2022",
    mainStack: [
      "html",
      "css",
      "js",
      "s3",
      "firebase",
      "android",
      "java",
      "kotlin",
      "ios",
      "swift-ui",
      "ad-manager",
      "aws",
    ],
    resume: true,
    resumeAccomplishments: [1, 2, 3],
    hasBlackLogo: false,
  },
  {
    key: "meaning-1",
    companyId: "meaning",
    companyName: "Meaning",
    startDate: "03-2022",
    endDate: "05-2023",
    mainStack: ["html", "css", "js", "bpmn", "gcp"],
    resume: true,
    resumeAccomplishments: [1, 2, 3],
    hasBlackLogo: false,
  },
  {
    key: "softplan-3",
    companyId: "softplan",
    companyName: "Softplan",
    startDate: "10-2020",
    endDate: "07-2021",
    mainStack: [
      "react",
      "micro-frontend",
      "dotnet",
      "graphql",
      "delphi",
      "sql-server",
      "aws",
    ],
    resume: false,
    resumeAccomplishments: [1, 2, 3],
    hasBlackLogo: false,
  },
  {
    key: "softplan-2",
    companyId: "softplan",
    companyName: "Softplan",
    startDate: "09-2019",
    endDate: "10-2020",
    mainStack: [
      "react",
      "micro-frontend",
      "dotnet",
      "graphql",
      "sql-server",
      "aws",
    ],
    resume: false,
    resumeAccomplishments: [1, 2, 4],
    hasBlackLogo: false,
  },
  {
    key: "softplan-1",
    companyId: "softplan",
    companyName: "Softplan",
    startDate: "12-2018",
    endDate: "08-2019",
    mainStack: [
      "react",
      "micro-frontend",
      "dotnet",
      "graphql",
      "sql-server",
      "aws",
    ],
    resume: false,
    resumeAccomplishments: [1, 2, 3],
    hasBlackLogo: false,
  },
  {
    key: "hotmart-3",
    companyId: "hotmart",
    companyName: "Hotmart",
    startDate: "06-2018",
    endDate: "11-2018",
    mainStack: ["vue", "nuxt", "node", "mysql", "html", "css", "js", "aws"],
    resume: false,
    resumeAccomplishments: [1, 2],
    hasBlackLogo: false,
  },
  {
    key: "hotmart-2",
    companyId: "hotmart",
    companyName: "Hotmart",
    startDate: "06-2017",
    endDate: "05-2018",
    mainStack: [
      "vue",
      "nuxt",
      "node",
      "mysql",
      "html",
      "css",
      "js",
      "php",
      "wordpress",
      "nginx",
      "aws",
    ],
    resume: false,
    resumeAccomplishments: [1, 2],
    hasBlackLogo: false,
  },
  {
    key: "hotmart-1",
    companyId: "hotmart",
    companyName: "Hotmart",
    startDate: "12-2016",
    endDate: "05-2017",
    mainStack: [
      "html",
      "css",
      "js",
      "php",
      "wordpress",
      "nginx",
      "mysql",
      "aws",
    ],
    resume: false,
    resumeAccomplishments: [],
    hasBlackLogo: false,
  },
  {
    key: "sydle-2",
    companyId: "sydle",
    companyName: "Sydle",
    startDate: "04-2014",
    endDate: "11-2016",
    mainStack: [
      "node",
      "express",
      "loopback",
      "angular",
      "html",
      "css",
      "js",
      "php",
      "wordpress",
      "magento",
      "apache",
      "bpmn",
      "linux",
      "windows",
      "oracle",
      "sql-server",
      "mysql",
      "aws",
    ],
    resume: false,
    resumeAccomplishments: [],
    hasBlackLogo: true,
  },
  {
    key: "sydle-1",
    companyId: "sydle",
    companyName: "Sydle",
    startDate: "04-2012",
    endDate: "03-2014",
    mainStack: [
      "js",
      "bpmn",
      "linux",
      "windows",
      "oracle",
      "sql-server",
      "aws",
    ],
    resume: false,
    resumeAccomplishments: [],
    hasBlackLogo: true,
  },
];
