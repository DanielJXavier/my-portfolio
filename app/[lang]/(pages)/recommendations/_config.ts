export const recommendations = [
  {
    authorId: "1758032899095",
    authorName: "Ben Lechlitner",
    company: "Lovevery",
  },
  {
    authorId: "1679942199683",
    authorName: "Nayane Menezes",
    company: "AB InBev",
  },
  {
    authorId: "1719021291674",
    authorName: "Wallace Freitas",
    company: "NEXT SI",
  },
  {
    authorId: "1716417055984",
    authorName: "Renan Silva",
    company: "Acert Grup",
  },
  {
    authorId: "1665746687624",
    authorName: "Camilla Polli",
    company: "OLX",
  },
  {
    authorId: "1711736735188",
    authorName: "Fernanda Rabelo",
    company: "OLX",
  },
  {
    authorId: "1692061488021",
    authorName: "Matheus Queiroz",
    company: "OLX",
  },
  {
    authorId: "1695594745530",
    authorName: "Douglas Ferreira",
    company: "OLX",
  },
  {
    authorId: "1565735422940",
    authorName: "Alan Cezar",
    company: "OLX",
  },
  {
    authorId: "1593552193665",
    authorName: "Matheus Missaci",
    company: "OLX",
  },
  {
    authorId: "1660079652443",
    authorName: "Ricardo Moraes",
    company: "VendorSmart",
  },
  {
    authorId: "1622852709278",
    authorName: "Mateus Queiroz Correia",
    company: "Nubank",
  },
  {
    authorId: "1606086729239",
    authorName: "Fernanda Ribeiro Durbano",
    company: "OLX",
  },
  {
    authorId: "1683719518610",
    authorName: "Gustavo Chimenes Dias",
    company: "OLX",
  },
  {
    authorId: "1712709978402",
    authorName: "CÃ¡ssio Fonseca",
    company: "Linx",
  },
  {
    authorId: "1695221396520",
    authorName: "Pedro Dias",
    company: "OLX",
  },
  {
    authorId: "1700516452017",
    authorName: "Euler Filho",
    company: "Techifide Ltd",
  },
  {
    authorId: "1713311445833",
    authorName: "Bruno Cechet",
    company: "Altruist",
  },
] as const;
