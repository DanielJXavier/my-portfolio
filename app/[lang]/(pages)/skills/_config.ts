export interface SkillInterface {
  key: string;
  name: string;
  size: 1 | 2 | 3 | 4 | 5;
}

export const hardSkills: SkillInterface[] = [
  /* Front-End */
  { key: "html", name: "HTML5", size: 5 },
  { key: "css", name: "CSS3", size: 5 },
  { key: "js", name: "JavaScript", size: 5 },
  { key: "react", name: "React", size: 4 },
  { key: "next", name: "Next.js", size: 3 },
  { key: "tailwind", name: "Tailwind CSS", size: 4 },
  { key: "vue", name: "Vue.js", size: 1 },
  { key: "nuxt", name: "Nuxt.js", size: 1 },
  { key: "angular", name: "Angular", size: 1 },
  { key: "responsive-design", name: "Responsive Design", size: 4 },
  { key: "seo", name: "SEO", size: 3 },
  { key: "ssr", name: "SSR", size: 2 },
  { key: "sass", name: "Sass", size: 4 },
  { key: "semantical-html", name: "Semantical HTML", size: 4 },
  { key: "ts", name: "TypeScript", size: 4 },
  { key: "webpack", name: "Webpack", size: 1 },
  { key: "redux", name: "Redux", size: 1 },
  { key: "jquery", name: "jQuery", size: 1 },
  { key: "acessibility", name: "Web Accessibility", size: 3 },
  { key: "web-vitals", name: "Core Web Vitals", size: 4 },
  { key: "performance", name: "Web Performance", size: 2 },
  { key: "security", name: "Web Security", size: 2 },
  { key: "web-sockets", name: "Web Sockets", size: 1 },
  { key: "styled-components", name: "Styled Components", size: 1 },
  { key: "pwa", name: "PWA", size: 2 },
  /* Back-End */
  { key: "node", name: "Node.js", size: 3 },
  { key: "express", name: "Express", size: 3 },
  { key: "loopback", name: "LoopBack", size: 1 },
  { key: "meteor", name: "Meteor", size: 1 },
  /* Test  */
  { key: "jest", name: "Jest", size: 3 },
  { key: "coverage", name: "Code Coverage", size: 3 },
  { key: "react-testing-library", name: "React Testing Library", size: 3 },
  { key: "cypress", name: "Cypress", size: 2 },
  { key: "chromatic", name: "Chromatic", size: 1 },
  /* APIs */
  { key: "http", name: "HTTP", size: 3 },
  { key: "graphql", name: "GraphQL", size: 1 },
  { key: "rest", name: "REST", size: 1 },
  /* PHP */
  { key: "php", name: "PHP", size: 1 },
  { key: "wordpress", name: "WordPress", size: 1 },
  { key: "magento", name: "Magento", size: 1 },
  { key: "laravel", name: "Laravel", size: 1 },
  /* Web Servers */
  { key: "nginx", name: "Nginx", size: 3 },
  { key: "apache", name: "Apache", size: 2 },
  /* CI / CD */
  { key: "github-actions", name: "GitHub Actions", size: 2 },
  { key: "bash", name: "Bash", size: 1 },
  { key: "docker", name: "Docker", size: 1 },
  { key: "yml", name: "YML", size: 1 },
  { key: "aws", name: "AWS", size: 2 },
  { key: "k8s", name: "Kubernets", size: 1 },
  { key: "terraform", name: "Terraform", size: 1 },
  /* Database */
  { key: "sql", name: "SQL", size: 3 },
  { key: "mysql", name: "MySQL", size: 1 },
  { key: "sql-server", name: "SQL Server", size: 1 },
  { key: "postgre", name: "PostgreSQL", size: 1 },
  { key: "oracle", name: "Oracle", size: 1 },
  { key: "nosql", name: "NoSQL", size: 2 },
  { key: "mongo", name: "MongoDB", size: 1 },
  /* Management Tools */
  { key: "jira", name: "Jira", size: 1 },
  { key: "confluence", name: "Confluence", size: 1 },
  { key: "slack-workflos", name: "Slack Workflows", size: 1 },
  { key: "slack-apps", name: "Slack Apps", size: 1 },
  { key: "bpmn", name: "BPMN", size: 1 },
  /* Business */
  { key: "a_b-testing", name: "A/B Testing", size: 1 },
  { key: "atomic-design", name: "Atomic Design", size: 1 },
  { key: "design-systems", name: "Design Systems", size: 1 },
  { key: "semver", name: "Semantic Versioning", size: 1 },
  { key: "ui_ux", name: "UI/UX", size: 1 },
  /* Other */
  { key: "functional-programming", name: "Functional Programming", size: 1 },
  { key: "oop", name: "OOP", size: 1 },
  { key: "regex", name: "Regular Expressions", size: 2 },
  { key: "markdown", name: "MarkDown", size: 1 },
];

export const softSkills = [
  { key: "communication", size: 5 },
  { key: "leadership", size: 5 },
  { key: "problem-solving", size: 5 },
  { key: "decision-making", size: 5 },
  { key: "empathy", size: 5 },
  { key: "conflict-resolution", size: 4 },
  { key: "adaptability", size: 4 },
  { key: "time-management", size: 3 },
  { key: "organization", size: 3 },
  { key: "strategic-thinking", size: 3 },
  { key: "mentoring", size: 4 },
  { key: "negotiation", size: 4 },
  { key: "resilience", size: 5 },
  { key: "critical-thinking", size: 3 },
  { key: "creativity", size: 3 },
  { key: "active-listening", size: 3 },
  { key: "team-building", size: 2 },
  { key: "ethical-decision-making", size: 1 },
  { key: "customer-orientation", size: 3 },
  { key: "learning-agility", size: 3 },
  { key: "curiosity", size: 4 },
  { key: "patience", size: 1 },
  { key: "public-speaking", size: 1 },
  { key: "networking", size: 1 },
  { key: "multitasking", size: 1 },
  { key: "attention-to-detail", size: 1 },
  { key: "motivation", size: 1 },
  { key: "persuasion", size: 1 },
  { key: "emotional-intelligence", size: 1 },
] as const;
