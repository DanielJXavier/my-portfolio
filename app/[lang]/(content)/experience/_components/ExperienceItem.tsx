import { useContext } from "react";

import Image from "next/image";

import { useParams } from "next/navigation";

import { getDictionary } from "get-dictionary";
import { Locale } from "i18n-config";

import { ExperienceModeContext } from "../Experience";

type ExperienceItemPropsType = Readonly<{
  role: string;
  companyId: string;
  companyName: string;
  year: string;
  description: string;
  responsibilities: string[];
  biggestChallenge: string;
  hasBlackLogo?: boolean;
  resume?: boolean;
}>;

type MapCompanyIdToColorType = {
  [key: string]: string;
};

const mapCompanyIdToColor: MapCompanyIdToColorType = {
  olx: "before:from-[#6d0ad5] before:via-[#6d0ad5]",
  meaning: "before:from-[#8568ec] before:via-[#8568ec]",
  softplan: "before:from-[#5d5ef1] before:via-[#5d5ef1]",
  hotmart: "before:from-[#ee4e21] before:via-[#ee4e21]",
  sydle: "before:from-secondary before:via-secondary",
};

export default function ExperienceItem({
  role,
  companyId,
  companyName,
  year,
  description,
  responsibilities,
  biggestChallenge,
  hasBlackLogo,
}: ExperienceItemPropsType) {
  const { lang } = useParams<{ lang: Locale }>();

  const {
    experience: { responsibilitiesTitle, biggestChallengeTitle },
  } = getDictionary(lang);

  const experienceMode = useContext(ExperienceModeContext);

  return (
    <article
      className={`my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b ${mapCompanyIdToColor[companyId]} before:to-background before:-z-10`}
      data-testid="experience-item"
    >
      <header className="flex items-start gap-x-2">
        <Image
          className={`rounded-full ${
            hasBlackLogo ? "border-2 border-solid border-secondary" : ""
          }`}
          src={`/experience/${companyId}.jpg`}
          width={40}
          height={40}
          alt={`Logo of ${companyName} company`}
        />
        <h2 className="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight">
          {role} @ {companyName}
          <span className="text-xs md:text-sm"> ({year})</span>
        </h2>
      </header>
      <section className="pl-14 lg:px-14">
        <h3 className="xl:text-xl">{description}</h3>
      </section>
      {experienceMode === "detailed" && (
        <>
          <section className="mt-4 pl-14 xl:px-14">
            <h3 className="font-semibold xl:text-xl">
              {responsibilitiesTitle}:
            </h3>
            <ul className="pl-7 list-disc text-sm xl:text-base">
              {responsibilities.map((responsibility, i) => (
                <li key={i}>{responsibility}</li>
              ))}
            </ul>
          </section>
          <section className="mt-4 pl-14 xl:px-14">
            <h3 className="font-semibold xl:text-xl">
              {biggestChallengeTitle}:
            </h3>
            <p className="text-sm xl:text-base">{biggestChallenge}</p>
          </section>
        </>
      )}
    </article>
  );
}
