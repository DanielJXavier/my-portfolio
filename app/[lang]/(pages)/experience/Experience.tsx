"use client";

import { createContext, useRef, useState } from "react";

import { useParams } from "next/navigation";

import { ExperienceItemsType, getDictionary } from "get-dictionary";
import { Lang } from "i18n-config";

import Title from "@/_components/Title";
import Work from "@/_icons/Work";

import Menu from "../_components/Menu";

import ExperienceItem from "./_components/ExperienceItem";
import { experience } from "./_config";

export const ExperienceModeContext = createContext("");

export default function Experience() {
  const { lang } = useParams<{ lang: Lang }>();

  const {
    experience: { title, menu, items, imageAltText },
  } = getDictionary(lang);

  const [experienceMode, setExperienceMode] = useState("simple");

  const menuItems = useRef([
    { text: menu.simple, value: "simple" },
    { text: menu.detailed, value: "detailed" },
  ]);

  const experienceItems = useRef<ExperienceItemsType>(items);

  return (
    <ExperienceModeContext.Provider value={experienceMode}>
      <Title icon={<Work />}>{title}</Title>

      <Menu
        items={menuItems.current}
        activeItem={experienceMode}
        handleClick={setExperienceMode}
      />

      {experience
        .filter(({ hidden }) => !hidden)
        .map(
          ({
            key,
            companyId,
            companyName,
            startDate,
            endDate,
            mainStack,
            hasBlackLogo,
          }) => (
            <ExperienceItem
              key={key}
              role={experienceItems.current[key].role}
              companyId={companyId}
              companyName={companyName}
              startDate={startDate}
              endDate={endDate}
              accomplishments={experienceItems.current[key].accomplishments}
              mainStack={mainStack}
              imageAltText={imageAltText}
              hasBlackLogo={hasBlackLogo}
            />
          )
        )}
    </ExperienceModeContext.Provider>
  );
}
