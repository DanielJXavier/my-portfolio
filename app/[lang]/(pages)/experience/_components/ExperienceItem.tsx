import { useContext } from "react";

import Image from "next/image";

import { useParams } from "next/navigation";

import { getDictionary } from "get-dictionary";
import { Lang } from "i18n-config";

import { hardSkills } from "../../skills/_config";

import { ExperienceInterface } from "../_config";
import { ExperienceModeContext } from "../Experience";

type ExperienceConfigType = Omit<
  ExperienceInterface,
  "key" | "resume" | "resumeAccomplishments"
>;
interface ExperienceItemPropsInterface extends ExperienceConfigType {
  role: string;
  description: string;
  accomplishments: string[];
  biggestChallenge: string;
  imageAltText: {
    prefix: string;
    sufix: string;
  };
}

type MapCompanyIdToColorType = {
  [key: string]: string;
};

const mapCompanyIdToColor: MapCompanyIdToColorType = {
  danielxavier: "before:from-secondary before:via-secondary",
  olx: "before:from-[#6d0ad5] before:via-[#6d0ad5]",
  meaning: "before:from-[#8568ec] before:via-[#8568ec]",
  softplan: "before:from-[#5d5ef1] before:via-[#5d5ef1]",
  hotmart: "before:from-[#ee4e21] before:via-[#ee4e21]",
  sydle: "before:from-secondary before:via-secondary",
};

export default function ExperienceItem({
  role,
  companyId,
  companyName,
  year,
  description,
  accomplishments,
  biggestChallenge,
  mainStack,
  imageAltText,
  hasBlackLogo,
}: ExperienceItemPropsInterface) {
  const { lang } = useParams<{ lang: Lang }>();

  const {
    experience: { accomplishmentsTitle, biggestChallengeTitle, mainStackTitle },
  } = getDictionary(lang);

  const experienceMode = useContext(ExperienceModeContext);

  return (
    <article
      className={`my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b ${mapCompanyIdToColor[companyId]} before:to-background before:-z-10`}
      data-testid="experience-item"
    >
      <header className="flex items-start gap-x-2">
        <Image
          className={`rounded-full ${
            hasBlackLogo ? "border-2 border-solid border-secondary" : ""
          }`}
          src={`/images/experience/${companyId}.jpg`}
          width={40}
          height={40}
          alt={`${imageAltText.prefix}${companyName}${imageAltText.sufix}`}
        />
        <h2 className="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight">
          {role} @ {companyName}
          <span className="text-xs md:text-sm"> ({year})</span>
        </h2>
      </header>
      <section className="pl-14 lg:px-14">
        <h3 className="xl:text-xl">{description}</h3>
      </section>
      {experienceMode === "detailed" && (
        <>
          <section className="mt-4 pl-14 xl:px-14">
            <h3 className="font-semibold xl:text-xl">
              {accomplishmentsTitle}:
            </h3>
            <ul className="pl-7 list-disc text-sm xl:text-base">
              {accomplishments.map((accomplishment, i) => (
                <li key={`accomplishment-${i}`}>{accomplishment}</li>
              ))}
            </ul>
          </section>
          <section className="mt-4 pl-14 xl:px-14">
            <h3 className="font-semibold xl:text-xl">
              {biggestChallengeTitle}:
            </h3>
            <p className="text-sm xl:text-base">{biggestChallenge}</p>
          </section>
          <section className="mt-4 pl-14 xl:px-14">
            <h3 className="font-semibold xl:text-xl">{mainStackTitle}:</h3>
            <p className="text-sm xl:text-base">
              {hardSkills
                .filter(({ key }) => mainStack.some((stack) => stack === key))
                .map(({ name }) => name)
                .join(" | ")}
            </p>
          </section>
        </>
      )}
    </article>
  );
}
