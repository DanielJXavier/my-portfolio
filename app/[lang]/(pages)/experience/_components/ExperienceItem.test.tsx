import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";

import ExperienceItem from "./ExperienceItem";

import { ExperienceModeContext } from "../Experience";

import { experience } from "../_config";
import { experience as experienceStrings } from "dictionaries/en.json";

import { useParams } from "next/navigation";

jest.mock("next/navigation");

(useParams as jest.Mock).mockImplementation(() => ({ lang: "en" }));

describe("ExperienceItem component", () => {
  it("renders the component", () => {
    const { key, companyId, companyName, year, hasBlackLogo } = experience[0];

    render(
      <ExperienceItem
        role={experienceStrings.items[key].role}
        companyId={companyId}
        companyName={companyName}
        year={year}
        description={experienceStrings.items[key].description}
        accomplishments={experienceStrings.items[key].accomplishments}
        biggestChallenge={experienceStrings.items[key].biggestChallenge}
        imageAltText={experienceStrings.imageAltText}
        hasBlackLogo={hasBlackLogo}
      />
    );

    const experienceItem = screen.getByTestId("experience-item");
    const image = experienceItem.querySelector("header > img");
    const title = experienceItem.querySelector("header > h2");
    const titleSpan = experienceItem.querySelector("header > h2 > span");
    const desc = experienceItem.querySelector("section > h3");

    expect(experienceItem).toBeInTheDocument();
    expect(image?.getAttribute("src")).toContain(companyId);
    expect(image?.getAttribute("alt")).toContain(companyName);
    expect(title?.innerHTML).toContain(experienceStrings.items[key].role);
    expect(title?.innerHTML).toContain(companyName);
    expect(titleSpan?.innerHTML).toContain(year);
    expect(desc?.innerHTML).toContain(experienceStrings.items[key].description);
  });

  it("renders the component with black logo style", () => {
    const { key, companyId, companyName, year } = experience[0];

    render(
      <ExperienceItem
        role={experienceStrings.items[key].role}
        companyId={companyId}
        companyName={companyName}
        year={year}
        description={experienceStrings.items[key].description}
        accomplishments={experienceStrings.items[key].accomplishments}
        biggestChallenge={experienceStrings.items[key].biggestChallenge}
        imageAltText={experienceStrings.imageAltText}
        hasBlackLogo={true}
      />
    );

    const experienceItem = screen.getByTestId("experience-item");

    expect(experienceItem.querySelector("header > img")?.className).toContain(
      "border-2 border-solid border-secondary"
    );
  });

  it("renders the component in the detailed mode", () => {
    const { key, companyId, companyName, year, hasBlackLogo } = experience[0];

    render(
      <ExperienceModeContext.Provider value="detailed">
        <ExperienceItem
          role={experienceStrings.items[key].role}
          companyId={companyId}
          companyName={companyName}
          year={year}
          description={experienceStrings.items[key].description}
          accomplishments={experienceStrings.items[key].accomplishments}
          biggestChallenge={experienceStrings.items[key].biggestChallenge}
          imageAltText={experienceStrings.imageAltText}
          hasBlackLogo={hasBlackLogo}
        />
      </ExperienceModeContext.Provider>
    );

    const experienceItem = screen.getByTestId("experience-item");

    const firstAccomplishmentsElement = experienceItem.querySelector(
      "section:nth-of-type(2) > ul > li"
    );

    const challenge = experienceItem.querySelector(
      "section:nth-of-type(3) > p"
    );

    expect(experienceItem).toBeInTheDocument();

    expect(firstAccomplishmentsElement?.innerHTML).toContain(
      experienceStrings.items[key].accomplishments[0]
    );

    expect(challenge?.innerHTML).toContain(
      experienceStrings.items[key].biggestChallenge
    );
  });
});
