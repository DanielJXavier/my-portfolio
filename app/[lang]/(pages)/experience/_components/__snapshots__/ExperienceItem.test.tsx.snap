// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ExperienceItem component renders the component 1`] = `
<div>
  <article
    class="my-6 sm:my-10 relative before:content-[''] before:absolute before:top-[40px] before:bottom-[-21px] before:sm:bottom-[-36px] last-of-type:before:bottom-0 before:left-[16px] before:ml-0.5 before:w-1 before:bg-gradient-to-b before:from-secondary before:via-secondary before:to-background before:-z-10"
    data-testid="experience-item"
  >
    <header
      class="flex items-start gap-x-2"
    >
      <img
        alt="Logo of PDSAcademy company"
        class="rounded-full border-2 border-solid border-secondary"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fexperience%2Fpdsacademy.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Fexperience%2Fpdsacademy.jpg&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fexperience%2Fpdsacademy.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <h2
        class="text-base sm:text-lg md:text-xl xl:text-2xl font-bold leading-tight"
      >
        Post-graduation professor
         @ 
        PDSAcademy
        <span
          class="text-xs md:text-sm"
        >
           
          (
          06-2025
           
          to
           
          06-2025
          )
        </span>
      </h2>
    </header>
    <section
      class="pl-14 lg:px-14"
    >
      <h3
        class="xl:text-xl"
      >
        A short-term contract role where I teach Front-End development.
      </h3>
    </section>
  </article>
</div>
`;
